scim-server/scim-server-common/src/main/java/edu/psu/swe/scim/server/rest/BaseResourceTypeResourceImpl.java

```diff
@@ -201,8 +201,8 @@ public Response query(AttributeReferenceListWrapper attributes, AttributeReferen
                                                 .map(wrapper -> wrapper.getAttributeReferences())
                                                 .orElse(Collections.emptySet()));
 
-    searchRequest.setFilter(filter.getFilter());
-
+    searchRequest.setFilter((filter != null) ? filter.getFilter() : null);
```

Comment:
Please add test cases for this NPE ... I'm not a fan of using the ternary form but working code wins.